server/src/internalClusterTest/java/org/elasticsearch/health/HealthMetadataServiceIT.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health;
+
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
+import org.elasticsearch.cluster.routing.allocation.DiskThresholdSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.health.metadata.HealthMetadata;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.junit.Before;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.test.NodeRoles.onlyRoles;
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class HealthMetadataServiceIT extends ESIntegTestCase {
+
+    private volatile boolean percentageMode;
+
+    @Before
+    public void setup() {
+        percentageMode = randomBoolean();
+    }
+
+    public void testEachMasterPublishesTheirThresholds() throws Exception {
+        try (InternalTestCluster internalCluster = internalCluster()) {
+            int numberOfNodes = 3;
+            Map<String, String> watermarkByNode = new HashMap<>();
+            for (int i = 0; i < numberOfNodes; i++) {
+                String customWatermark = percentageMode
+                    ? randomIntBetween(60, 80) + "%"
+                    : new ByteSizeValue(randomIntBetween(10, 100)).toString();
+                String nodeName = startNode(internalCluster, customWatermark);
+                watermarkByNode.put(nodeName, customWatermark);
+            }
+            ensureStableCluster(numberOfNodes);
+
+            String electedMaster = internalCluster.getMasterName();
+            {
+                HealthMetadata.DiskThresholds.DiskThreshold lowWatermark = HealthMetadata.getHealthCustomMetadata(
```

Comment:
naming game :) 

This is a bit difficult to parse (at least to me) - `HealthMetadata.DiskThresholds.DiskThreshold`

Since this is in the health metadata could (should?) we come up with a new concept? Alert limits? Health limits? Disk health limits?

Reply:
I see, I have to admit that threshold sounds better than limits but I will play around with naming see if we can come up with something better.

Reply:
What about `HealthMetadata` contains `DiskMetadata` (for now it's mainly thresholds, but you never know what else we might need) and then we have the `DiskThreshold` I could also go for `DiskHealthLimits` but I am not sure if it is easier to read or not. 

About `HealthLimits` or `AlertLimits` I think it's too generic and we will have issues when we introduce other resource metadata.

Another idea I had, and I am leaning towards this tbh, is to use the `RelativeByteSizeValue` instead of `DiskThreshold`. What do you think?

I am open to other ideas :)

Reply:
How about `HealthMetadata.Disk.Threshold` ? (the FQN heps removing the extra "metadata" and "disk" names)

> Another idea I had, and I am leaning towards this tbh, is to use the RelativeByteSizeValue instead of DiskThreshold. What do you think?

Are you thinking of naming here? Or the actual `RelativeByteSizeValue` entity instead of the DiskThreshold entity?

Reply:
I was thinking to use the actual `RelativeByteSizeValue` java entity since we are serializing it via string conversion. 

> How about HealthMetadata.Disk.Threshold

I do like this! I wanted to avoid generic names because it might confuse people when searching in the IDE. But since it's an inner class and it looks quite nice... I am reconsidering :) Shall we go for this?

Reply:
I'm a fan :) If you like it too I say we go for this (@dakrone might have a different opinion but we can incorporate those too if they come through :) )

Reply:
I want for `HealthMetadata.Disk.Threshold` :)

Reply:
@andreidan So, I am reconsidering this choice since distributed team will be working soon(-ish) on #81406. I am proposing to already use `RelativeByteSizeValue` instead of the `Threshold`. The reason is that it has the `calculateValue` method which calculates the percentage and takes into account the `headroom`, so for the frozen tier it will take the headroom into account and for the others we will use `-1` for now until the headroom is implemented. This will make easier the transition when headroom  is implemented everywhere. Any objections?

Reply:
Proposal applied